### PR TITLE
508 compliance for taxa/taxonomy/taxonomydynamicdisplay.php

### PR DESCRIPTION
**Pull request by NikitaSalikov**
## Changes Made:

- Added Doctype and lang attrbute
- Fixed deprecated FILTER SANITIZE...
- Added styling to css for img since border is deprecated. Also added stylings and changed fonts to relative for Central Thesaurus
- Fixed the search bar in mobile version
- Added labels to search bar and checkboxes.   

## Pull Request Checklist:

- [X] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [N/A] Hotfixes should be merged into both the `Development` and `master` branches (at the same time).
- [X] All new text is preferrably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages)
- [X] There are no linter errors
- [X] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [X] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [N/A] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [X] Comment which GitHub issue(s), if any does this PR address
- [X] It is the code author's responsibility to merge their own pull request after it has been approved
- [X] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [N/A] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [N/A] If the dev team has agreed that this PR represents the last PR going into the Development branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place

Thanks for contributing and keeping it clean!

### CODE REVIEW

**Comment by NikitaSalikov** [comment]
Mark, could you please run the Total Validator on this updated page? I'd like to see what issues remained and which ones were not related to the page. Thank you!

File: taxa/taxonomy/taxonomydynamicdisplay.php [hunk 1]
```diff
@@ -101,7 +103,7 @@ function displayTaxomonyMeta(){
 			?>
 			<div style="float:right;" title="<?php echo (isset($LANG['ADD_NEW_TAXON'])?$LANG['ADD_NEW_TAXON']:'Add a New Taxon'); ?>">
 				<a href="taxonomyloader.php" target="_blank">
-					<img style='border:0px;width:15px;' src='../../images/add.png'/>
+					<img class="img-add" src="../../images/add.png" alt="Add a new Taxon">
```
Comment (Atticus29) [inline taxa/taxonomy/taxonomydynamicdisplay.php]: We should start internationalizing the alt attributes as well.

File: taxa/taxonomy/taxonomydynamicdisplay.php [hunk 2]
```diff
@@ -126,20 +128,25 @@ function displayTaxomonyMeta(){
 		</div>
 		<div style="clear:both;">
 			<form id="tdform" name="tdform" action="taxonomydynamicdisplay.php" method='POST'>
-				<fieldset style="padding:10px;width:500px;">
+				<fieldset style="padding:10px; max-width:500px;">
```
Comment (Atticus29) [inline taxa/taxonomy/taxonomydynamicdisplay.php]: If possible, move this styling to a .css file.

File: taxa/taxonomy/taxonomydynamicdisplay.php [hunk 3]
```diff
@@ -126,20 +128,25 @@ function displayTaxomonyMeta(){
...
+						<label for="taxontarget"> <?php echo htmlspecialchars($LANG['TAXON'], HTML_SPECIAL_CHARS_FLAGS) ?>: </label>
+						<input id="taxontarget" name="target" type="text" class="taxon-search-bar" value="<?php echo $taxonDisplayObj->getTargetStr(); ?>" />
 					</div>
 					<div style="float:right;margin:15px 80px 15px 15px;">
 						<button name="tdsubmit" type="submit" value="displayTaxonTree"><?php echo (isset($LANG['DISP_TAX_TREE'])?$LANG['DISP_TAX_TREE']:'Display Taxon Tree'); ?></button>
 						<input name="taxauthid" type="hidden" value="<?php echo $taxAuthId; ?>" />
 					</div>
 					<div style="margin:15px 15px 0px 60px;">
-						<input name="displayauthor" type="checkbox" value="1" <?php echo ($displayAuthor?'checked':''); ?> /> <?php echo (isset($LANG['DISP_AUTHORS'])?$LANG['DISP_AUTHORS']:'Display authors'); ?>
+						<input name="displayauthor" type="checkbox" value="1" <?php echo ($displayAuthor?'checked':''); ?> />
```
Comment (Atticus29) [inline taxa/taxonomy/taxonomydynamicdisplay.php]: this `<input>` element needs an `id` attribute for the label to work correctly.

**Review by Atticus29 — COMMENTED** [review]
Just a few small changes needed.

In addition to the ones commented in the code, remove
the `type` attribute from `<style type="text/css">`.

File: taxa/taxonomy/taxonomydynamicdisplay.php [hunk 4]
```diff
@@ -101,9 +101,9 @@ function displayTaxomonyMeta(){
 		}
 		if($isEditor){
 			?>
-			<div style="float:right;" title="<?php echo (isset($LANG['ADD_NEW_TAXON'])?$LANG['ADD_NEW_TAXON']:'Add a New Taxon'); ?>">
+			<div style="float:right;"?>
```
Comment (Atticus29) [inline taxa/taxonomy/taxonomydynamicdisplay.php]: `?` here is a typo.

**Review by Atticus29 — APPROVED** [review]
I pushed a few really easy changes. Should be ready for merge now. Thanks!

**Pull request merged**